pytradfri/device.py

```diff
@@ -125,8 +125,6 @@ def power_source(self):
     @property
     def power_source_str(self):
         """String representation of current power source."""
-        if DeviceInfo.ATTR_POWER_SOURCE not in self.raw:
-            return None
```

Comment:
This part of the code can never be reached, so it should be removed.

Reply:
This case can never occur?

Reply:
As far as I was able to determine it always returns `Unkown` from line 130 instead of `None`.

After replacing line 129 with `return 'ERR'` I got the following results:
```python
    dev = Device({})
    print(dev.device_info.power_source_str)  # KeyError: '3' from device.py:140

    dev = Device({'3': {}})
    print(dev.device_info.power_source_str)  # None

    dev = Device({'3': {'6': None}})
    print(dev.device_info.power_source_str)  # 'Unkown'

    dev = Device({'3': {'6': 'abc'}})
    print(dev.device_info.power_source_str)  # 'Unkown'

    dev = Device({'3': {'6': 1}})
    print(dev.device_info.power_source_str)  # 'Internal Battery'

    dev = Device({'3': {'6': True}})
    print(dev.device_info.power_source_str)  # 'Internal Battery'

    dev = Device({'3': {'6': [1, 2]}})
    print(dev.device_info.power_source_str)  # TypeError

    dev = Device({'3': {'6': {1, 2}}})
    print(dev.device_info.power_source_str)  # TypeError
```

`'ERR'` is never a result, thus that spot can't be reached.

Reply:
This was added by @balloob long backm surely for a reason. It might be that it's a part of the COAP-protocol. I'm not willing to remove the code at this stage.

Reply:
Very well, I'll reverse it. It's not a lot of bloat so it'll be fine.